python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,11 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_should_validate_input_args(self):
+        from pyspark.sql.functions import udf
+
+        self.assertRaises(TypeError, udf(lambda x: x), None)
```

Comment:
If this is covered by existing tests, then that's fine. Good point.

To validate number of args, I think it is a good idea, as long as we know that it won't fail C extensions (but may be inconclusive).